src/components/DownloadCsv/index.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Icon } from '@edx/paragon';
+import StatusAlert from '../../components/StatusAlert';
+
+
+class DownloadCsv extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      csvLoading: false,
+    };
+  }
+
+  renderDownloadCsvButton() {
+    const {
+      csvFetchMethod,
+      csvOptions,
+      enterpriseId,
+    } = this.props;
+
+    const downloadButtonIconClasses = this.state.csvLoading ? ['fa-spinner', 'fa-spin'] : ['fa-download'];
+
+    return (
+      <Button
+        className={['btn-outline-primary', 'download-btn']}
+        disabled={this.state.csvLoading}
+        label={
+          <span>
+            <Icon className={['fa', 'mr-2'].concat(downloadButtonIconClasses)} />
+            Download report (CSV)
+          </span>
+        }
+        onClick={() => csvFetchMethod(enterpriseId, csvOptions)}
```

Comment:
I'd recommend we wire this up to redux by changing the current actions and reducers in `/actions/courseEnrollments.js` and `/reducers/courseEnrollments.js` to be generic CSV, similar to how the table.js files are generic handlers for the table.

The `/actions/courseEnrollments.js` handler would then take an additional parameter of the fetchMethod and call that:

```
const fetchCsv = (enterpriseId, fetchCsvMethod) => (
  (dispatch) => {
    dispatch(fetchCsvRequest());
    return fetchCsvMethod(enterpriseId)
      .then((response) => {
        saveAs(new Blob([response.data]), `${enterpriseId}_progress_report.csv`);
        dispatch(fetchCsvSuccess());
      })
      .catch((error) => {
        dispatch(fetchCsvFailure(error));
      });
  }
);
```